src/Arrays/Programs_involving_Data_Sequences/Hard/Q1/MainTest.java

```diff
@@ -0,0 +1,65 @@
+package Arrays.Programs_involving_Data_Sequences.Hard.Q1;
+
+import global.BaseTest;
+import global.variables.*;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest extends BaseTest {
+    int[] inputs = {0, 1, 0, 2, 3, 1, 0, 2, 0};
+
+    public Clause[] testSentence() {
+        Clause[] output = new Clause[19];
+        for (int x = 0; x < 17; x += 2) {
+            output[x] = new IntegerLiteral(inputs[x / 2]);
+            output[x + 1] = new NewLine();
+        }
+        output[18] = new IntegerLiteral(3);
+        return output;
+    }
+
+    public void runMain() {
+        CountAr.main(new String[]{});
+    }
+
+    @RepeatedTest(50)
+    public void methodTestRandomNums() {
+        int[] expected = new int[10];
+        int[] passed = new int[10];
+        for (int x = 0; x < 10; x++) {
+            int n = (int) (Math.random() * 10);
+            passed[x] = n;
+            expected[n]++;
+        }
+        int[] actual = CountAr.countThisPlease(passed);
+
+        assertArrayEquals(expected, actual, "Your method did not correctly count for the array: " + Arrays.toString(passed));
+    }
+
+    @Test
+    public void methodOutOfBoundsException() {
+
+        try {
+            int[] hold = CountAr.countThisPlease(new int[]{10, 10, 10, 9, 2, 3, 4, 5, 1, 7, 5});
+            fail("Your program should not count numbers larger than 9");
+        } catch (IndexOutOfBoundsException e) {
+        }
+    }
+
+    @Test
+    public void methodPassingAnArrayLargerThan10() {
+        int[] expected = new int[10];
+        int[] passed = new int[30];
+        for (int x = 0; x < passed.length; x++) {
+            int n = (int) (Math.random() * 10);
+            passed[x] = n;
+            expected[n]++;
+        }
+        int[] actual = CountAr.countThisPlease(passed);
+        assertArrayEquals(expected, actual, "Your program did not correctly count all the numbers given in the input");
```

Comment:
Maybe make this a tad more specific like `Your program should only count arrays that have exactly 10 values` then you can even check for arrays that have less than 10 elements.

Reply:
I think there's been a slight misunderstanding, it's not supposed to fail if it counts more than 10 values. The program should be able to take an array of any length as long as the elements fall in the range of 0-9. I can write another test that checks under 10 elements doesn't crash.

Reply:
Oh my bad then. Try to check that the array is definitely 10 elements or over then, and fail otherwise. You might have to update the question to ensure that this requirement is communicated as well

Reply:
What exactly is the requirement? Are you thinking of adding a condition to what the array input length must be?

Reply:
Sorry about the misunderstanding. Upon looking at the question again I think your tests are fine. But I think you need to add the question type to the code? I'm not sure if this is a parsons or a Java

Reply:
I thought it was assumed if there wasn't a parsons label that it was java

Reply:
Yep you're right. I wrote this before the meeting so I was unsure.